Lib/test/test_asyncio/test_subprocess.py

```diff
@@ -184,6 +185,21 @@ def test_kill(self):
         else:
             self.assertEqual(-signal.SIGKILL, returncode)
 
+    @unittest.skipUnless(shutil.which('sleep'), 'sleep is not available')
+    def test_kill_issue43884(self):
+        blocking_shell_command = 'sleep 10000'
```

Comment:
Please run Python and run ``import time; time.sleep(...)`` to not rely on an external program.

Reply:
That would not work on Windows were killing a process leaves its directory locked and causes PermissionErrors.

Reply:
Interesting, I wasn't able to reproduce this bug on windows 11. I'll have to check again with your code.

Reply:
Here's my full code:
```
import asyncio

async def test():
    process = await asyncio.create_subprocess_shell(
        "timeout 5\necho done",
        stdout=asyncio.subprocess.PIPE,
    )
    await asyncio.sleep(1)
    process.kill()
    await process.wait()
    # process._transport.close()

asyncio.run(test())
```

I'm on Windows 11 too (though it's probably a Microsoft internal early release). With the above script I get this output:

<details>

```
Exception ignored in: <function BaseSubprocessTransport.__del__ at 0x00000155BF7564D0>
Traceback (most recent call last):
  File "C:\Users\gvanrossum\AppData\Local\Programs\Python\Python310\lib\asyncio\base_subprocess.py", line 126, in __del__
    self.close()
  File "C:\Users\gvanrossum\AppData\Local\Programs\Python\Python310\lib\asyncio\base_subprocess.py", line 104, in close
    proto.pipe.close()
  File "C:\Users\gvanrossum\AppData\Local\Programs\Python\Python310\lib\asyncio\proactor_events.py", line 108, in close
    self._loop.call_soon(self._call_connection_lost, None)
  File "C:\Users\gvanrossum\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 750, in call_soon
    self._check_closed()
  File "C:\Users\gvanrossum\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
Exception ignored in: <function _ProactorBasePipeTransport.__del__ at 0x00000155BF757F40>
Traceback (most recent call last):
  File "C:\Users\gvanrossum\AppData\Local\Programs\Python\Python310\lib\asyncio\proactor_events.py", line 115, in __del__
    _warn(f"unclosed transport {self!r}", ResourceWarning, source=self)
  File "C:\Users\gvanrossum\AppData\Local\Programs\Python\Python310\lib\asyncio\proactor_events.py", line 79, in __repr__
    info.append(f'fd={self._sock.fileno()}')
  File "C:\Users\gvanrossum\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_utils.py", line 102, in fileno
    raise ValueError("I/O operation on closed pipe")
ValueError: I/O operation on closed pipe
```

</details>

Reply:
I tried your code and indeed I can now reproduce the issue. I am only able to reproduce this on command prompt though, not with git bash which is what I use on Windows by changing `timeout` with `sleep`. This is has something to do with how cmd handles killing process vs `bash`.

Reply:
I pushed an update to use process group on Windows so that killing the process kills all its child processes.

Reply:
Output on Windows 11 after fix on cmd.exe:
```console
D:\python\main>python.bat main.py
Running Debug|x64 interpreter...
```